Final/LinkedList.h

```diff
@@ -0,0 +1,32 @@
+#ifndef LINKED_LIST_H
+#define LINKED_LIST_H
+
+typedef struct node {
+    int data;
+    struct node *next;
+} node;
+
+class LinkedList {
```

Comment:
in an attempt to really get you versed in c++, let's start namespacing everything to keep our code clean and not pollute the global namespace, and sorry, it looks like all standard libraries use snake_case for everything, so let's try to adhere to that as well.

namespaces are just another form of encapsulation in c++. you've already used the std:: namespace that wraps all standard c++ libraries, so let's define our own namespace for this code. maybe just `dlpf`? so this .h would become:

```
#ifndef LINKED_LIST_H
#define LINKED_LIST_H

namespace dlpf {
    typedef struct node {
        int data;
        struct node *next;
    } node;

    class linked_list {
        public:
        // Constructor
        linked_list();
        // Destructor
        ~linked_list();

        // Pushes node to front of list
        void push(int data);
        // Removes the first instance of a node with the given data
        void remove(int data);

        // Exposed to allow for easy iteration through the linked list:
        // for (node *cur = list.begin(); cur != list.end(); cur = cur->next) {
        //     // Do something
        // }
        node *begin();
        node *end();
    private:
        node *head;
        node *sentinel;
    };

    #endif 
} // dlpf
```
note that we should always mark the closing } of the namespace with a comment for clarity. in the .cpp, instead of defining your methods like `void linked_list::push(int data)`, you'd do `void dlpf::linked_list::push(int data)`